payment/banks/banks.py

```diff
@@ -0,0 +1,371 @@
+from abc import ABC, abstractmethod
+import logging
+import uuid
+from urllib import parse
+from django.db.models import Q
+from django.shortcuts import redirect
+from django.utils import timezone
+from django.urls import reverse
+#
+# from .. import default_settings as settings
+# from ..exceptions import CurrencyDoesNotSupport, AmountDoesNotSupport, BankGatewayTokenExpired, BankGatewayStateInvalid
+# from ..models import Bank, CurrencyEnum, PaymentStatus
+# from ..utils import append_querystring
+
+
+class BaseBank(ABC):
+    """Base bank for sending to gateway."""
+    # TODO واحد پول باید درست بشه
+    _gateway_currency: str = CurrencyEnum.IRR
+    _currency: str = CurrencyEnum.IRR
+
+    _amount: int = 0
+    _gateway_amount: int = 0
+    _mobile_number: str = None
+    _tracking_code: int = None
+    _reference_number: str = ''
+    _transaction_status_text: str = ''
```

Comment:
Use enum for defigning a status
we need a model that parses response status to our enum field